backend/server.js

```diff
@@ -42,8 +44,21 @@ sessionStore.sync();
 
 app.use(stockRoutes);
 app.use(userRoutes);
+
+const checkSession = (req, res, next) => {
+    const user = req.session.user;
+    if (user == null) {
+        res.status(401).json({ error: "Missing Session." });
+    } else {
+        next();
+    }
+};
+
+app.use(checkSession);
```

Comment:
Some routes don't need the session check. Is this working out because those routes are being used above checkSession?

Reply:
Yes, exactly.